### PR TITLE
Feature/replace comment creation bitbucket

### PR DESCRIPTION
**Pull request by IgorAlbert**
1. commit: needed a working solution, which would fix the problem, left a lot of useless code.
2. commit: was trying to find a smaller solution where i don't need 2 classes.

### CODE REVIEW

File: src/main/java/org/retest/rebazer/domain/BitbucketContent.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package org.retest.rebazer.domain;
+
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder
+public class BitbucketContent {
+	private String raw;
```
Comment (beatngu13) [inline src/main/java/org/retest/rebazer/domain/BitbucketContent.java]: If you use `@Value`, you can omit `private`.

File: src/main/java/org/retest/rebazer/domain/BitbucketContent.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
```
Comment (beatngu13) [inline src/main/java/org/retest/rebazer/domain/BitbucketContent.java]: Why this line break?

File: src/main/java/org/retest/rebazer/domain/BitbucketContent.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package org.retest.rebazer.domain;
+
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder
```
Comment (beatngu13) [inline src/main/java/org/retest/rebazer/domain/BitbucketContent.java]: The class is rather simple, is there a reason you have chosen `@Builder`? Why the class in the first place, isn't it just a wrapper for a string or is required somewhere else?

File: src/main/java/org/retest/rebazer/domain/BitbucketContent.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+public class BitbucketContent {
```
Comment (beatngu13) [inline src/main/java/org/retest/rebazer/domain/BitbucketContent.java]: Isn't it `BitbucketComment` or `BitbucketCommentContent`?

File: src/main/java/org/retest/rebazer/connector/BitbucketConnector.java [hunk 5]
```diff
@@ -124,10 +121,11 @@ private DocumentContext jsonPathForPath( final String urlPath ) {
 
 	@Override
 	public void addComment( final PullRequest pullRequest, final String message ) {
-		final Map<String, String> request = new HashMap<>();
-		request.put( "content", message );
+		final Map<String, BitbucketContent> request = new HashMap<>();
+		final BitbucketContent comment = BitbucketContent.builder().raw( message ).build();
+		request.put( "content", comment );
```
Comment (beatngu13) [inline src/main/java/org/retest/rebazer/connector/BitbucketConnector.java]: You could also include the map in `BitbucketComment`, something like:

```java
@Value
public class BitbucketComment {
    private static final String KEY = "content";
    String raw;
    Map<String, String> request() {
        final Map<String, String> request = new HashMap();
        request.put( KEY, raw );
        return request;
    }
}
```

Then you could do:

```java
BitbucketComment comment = new BitbucketComment( message );
template.postForObject( requestPath( pullRequest ) + "/comments", comment.request(), String.class );
```

However, @modulo11 is more familiar with Spring and its best practices.

File: src/main/java/org/retest/rebazer/domain/BitbucketContent.java (hunk 3 above)
Comment (IgorAlbert) [reply src/main/java/org/retest/rebazer/domain/BitbucketContent.java]: I needed it to get another nesting depth, however there was a simpler solution dismiss at this point

File: src/main/java/org/retest/rebazer/connector/BitbucketConnector.java (hunk 5 above)
Comment (IgorAlbert) [reply src/main/java/org/retest/rebazer/connector/BitbucketConnector.java]: I spoke to @modulo11 a little bit about it and we decided to solve this problem with a simpler solution without the Lombok library

File: src/main/java/org/retest/rebazer/connector/BitbucketConnector.java [hunk 6]
```diff
@@ -124,10 +121,12 @@ private DocumentContext jsonPathForPath( final String urlPath ) {
 
 	@Override
 	public void addComment( final PullRequest pullRequest, final String message ) {
-		final Map<String, String> request = new HashMap<>();
-		request.put( "content", message );
+		final ObjectNode contentNode = objectMapper.createObjectNode();
+		final ObjectNode contentNode1 = objectMapper.createObjectNode();
+		contentNode1.put( "raw", message );
+		contentNode.set( "content", contentNode1 );
```
Comment (beatngu13) [inline src/main/java/org/retest/rebazer/connector/BitbucketConnector.java]: I would suggest to change the variables names and statements order. For instance:

```java
final ObjectNode comment = objectMapper.createObjectNode();
comment.put( "raw", message );
final ObjectNode request = objectMapper.createObjectNode();
request.set( "content", comment );
```

If you can derive better names from the actual API, that's also good.

Comment (IgorAlbert) [reply src/main/java/org/retest/rebazer/connector/BitbucketConnector.java]: sry about that the names were somehow not adopted...

**Pull request merged**